PocketKit/Sources/PocketKit/MyList/ArchivedItemsList/ArchivedItemsListViewModel.swift

```diff
@@ -152,11 +163,54 @@ class ArchivedItemsListViewModel: ItemsListViewModel {
                 },
                 UIAlertAction(title: "Yes", style: .destructive) { [weak self] _ in
                     self?.presentedAlert = nil
+                    self?.delete(item: item)
                 }
             ],
             preferredAction: nil
         )
     }
+
+    private func delete(item: ArchivedItem) {
+        Task {
+            try? await source.delete(item: item)
+
+            guard let index = archivedItems.firstIndex(of: item) else {
+                return
+            }
+
+            archivedItems.remove(at: index)
+            _events.send(.snapshot(buildSnapshot()))
+        }
+    }
+
+    private func favorite(item: ArchivedItem) {
+        Task { await setIsFavorite(true, on: item) }
```

Comment:
Same question as above for some of these: do we have to wait for the call to finish?

Reply:
See above